### PR TITLE
Additional convenience for create-mountaineer-app projects

### PR DESCRIPTION
**Pull request by stefnnn**
When generating a new project with create-mountaineer-app, you most likely want to create a git repo. Having a preconfigured .gitignore, which also includes the automatically generated directories should be a benefit.

Also, I was bothered that when refactoring code there were always dozens of search results from generated js/ts files. Adding editor configurations for VS Code (and vim ;-)) could be beneficial. Let me know if you think this is too specific.

### CODE REVIEW

File: create_mountaineer_app/create_mountaineer_app/templates/.gitignore [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# General excludes
```
Comment (piercefreeman) [inline create_mountaineer_app/create_mountaineer_app/templates/.gitignore:1]: Looks like this is based on the github python base?
https://github.com/github/gitignore/blob/main/Python.gitignore

If so, mind dropping a link to the original in the gitignore header so users are aware of where this black magic config came from?

**Review by piercefreeman — COMMENTED** [review]
This looks good so far. I don't love the idea of bundling _all_ of the IDE specific configuration parameters for each project though. Will end up cluttering the default filesystem when most users only use one.

Instead I think a better fit might be to modify the initial CLI configuration and ask users what their preferred IDE is. You should be able to use the questionary [rawselect](https://questionary.readthedocs.io/en/stable/pages/types.html#type-checkbox) type to choose from a list of supported IDEs. I know I'm for sure going to add one for Zed as well. :)

Relevant code: https://github.com/piercefreeman/mountaineer/blob/main/create_mountaineer_app/create_mountaineer_app/cli.py

File: create_mountaineer_app/create_mountaineer_app/templates/.gitignore (hunk 1 above)
Comment (stefnnn) [reply create_mountaineer_app/create_mountaineer_app/templates/.gitignore]: There was a comment and link to the source already a bit further down, where the black magic starts. But I preferred to keep the project specific ignores at the top.

**Comment by stefnnn** [comment]
Thanks for the suggestions and the implementation hints, the editor configuration is now configurable (or skippable).

**Comment by piercefreeman** [comment]
Thanks for the changes. Tests look good, you might just need to run a `make lint` to clear out the linting errors raised by CI. After that I'd say good to merge.

**Pull request merged**